### PR TITLE
Implementation of variable step counts for continuous reprocessing

### PR DESCRIPTION
**Pull request by LukeSeifert**
## Overview
This PR primarily serves to allow for a variable number of Serpent2 steps to be included for each continuous reprocessing model. This is accomplished by looping and using the read values to read the previous material data. The plotting is then performed by a newly added plotting class in a separate file for readability, since otherwise it became too messy.

Closes issue #60

## Other Minor Changes

- Updated `type_1_flow` notes on different reprocessing approaches that are in the planning phase. I also added notes to it in the `user_input` file where I anticipate adding the options to enable or disable this run type.

- Removed multiplot / close_boolean option from `user_input`, `serpent_output`, and `run_user_input`. I removed this since I realized it should be part of the `serpent_plotting`, but it isn't incorporated yet. Likely a future PR.

### CODE REVIEW

File: msr-flows/Notes/type_1_flow.ipynb [hunk 1]
```diff
@@ -129,11 +129,117 @@
    ]
   },
   {
-   "cell_type": "code",
-   "execution_count": null,
+   "cell_type": "markdown",
```
Comment (yardasol) [inline msr-flows/Notes/type_1_flow.ipynb:132]: Great explanation!

File: msr-flows/msbr_compare/run_user_input.py [hunk 2]
```diff
@@ -448,70 +450,46 @@ def cycle_time_decay(self, identifier='CTD'):
                     label=SP_identifier,
                     alpha=ui.overlap,
                     lw=ui.width)
-            for each_step in range(ui.number_serp_steps):
```
Comment (yardasol) [inline msr-flows/msbr_compare/run_user_input.py:451]: Why did you remove this for loop?

**Review by yardasol — APPROVED** [review]
- [x] Read the PR description
- [x] Read through all the changes, considering the following questions.
  - [x] Is there anything you can praise about this PR? Start with
        praise.
  - [x] Are variable names brief but descriptive?
  - [x] Are new/changed functions no longer than a paragraph?
  - [x] Do all function parameters have default values where appropriate?
  - [x] Is the code clear and clean? (see Robert C. Martin's
        [Clean Code](https://i-share-uiu.primo.exlibrisgroup.com/discovery/fulldisplay?docid=alma99944155312205899&context=L&vid=01CARLI_UIU:CARLI_UIU&tab=LibraryCatalog&lang=en))
  - [x] Is there enough documentation?
  - [x] Does the programming style meet the requirements of the
        repository ([PEP8](https://www.python.org/dev/peps/pep-0008/) for python,
        [google C++ style guide](https://google.github.io/styleguide/cppguide.html), etc.)
  - [ ] If a new feature has been added, or a bug fixed, has a test been
        added to confirm good behavior?
  - [ ] Does the test actually test the new/changed functionality?
  - [ ] Does the test successfully test edge cases?
  - [ ] Does the test successfully test corner cases?
  - [ ] If the repository has continuous integration: does the PR pass
        the tests?
  - [ ] If there is no continuous integration, check out the branch
        locally, build, and run the tests.
  - [ ] Do the tests pass on your machine?
  - [x] Is the PR free of random cruft (built files, `.swp` files,
        etc.)?
  - [x] Do all files in the PR belong in the repository?
  - [ ] If the PR deletes files, is this appropriate?
  - [ ] If the PR adds files or data, are these new files compatible
        with the repository license?
  - [x] Make a review, leaving kind comments and suggesting changes
        where needed (to resolve the above).
  - [ ] Has the author resolved all of the comments and suggestions
        in your review?
- [ ] When you approve of the PR, merge and close it.
- [x] Does this PR close an issue? If so, be sure to descriptively
      close this issue when the PR is merged
- [x] Thank the author for their contribution.

Hi @LukeSeifert, great work on this! I love to see your docstrings looking up to PEP 257 standards :)

I just have one question about an algorithm change in one of your files. I made an inline comment about it. Otherwise, I think I'm gonna go ahead and approve this.

File: msr-flows/msbr_compare/run_user_input.py (hunk 2 above)
Comment (LukeSeifert) [reply msr-flows/msbr_compare/run_user_input.py:451]: Good question! I had noticed having that large outer loop was making things less organized, so I moved the loop into the `serpent_plotting` script functions if I decide to use the loop. Line 60 of that file contains this same line, but it made things more organized doing it this way I think.
